nncf/torch/dynamic_graph/context.py

```diff
@@ -91,9 +93,9 @@ class TracingContext:
     def __init__(self):
         self.graph = DynamicGraph()
 
-        self._post_hooks = {}
-        self._pre_hooks: Dict[PreHookId, List[Callable]] = {}
-        self._num_nested_hooks = 0
+        self._post_hooks = defaultdict(OrderedDict)
+        self._pre_hooks: Dict[PreHookId, List[Callable]] = defaultdict(OrderedDict)
+        self._hooks_counter = -1
```

Comment:
The first entered hook would be missed. Is it right?

Reply:
No. I would better call this `_hooks_hash_id`